### DISCUSSION TITLE
Low profile keys for LMN3

### DISCUSSION THREAD

**Discussion by brassmonk** [FundamentalFrequency/.github#9] [Ideas]
I wanted to see if anyone has experience with mechanical keyboards that could help modify the cad file to accept low profile key switches (specifically kailh choc switches) I love the over all design of this project but I just wanted to see if it would be possible to get a more minimalistic thinner aesthetic for travelling with it.

**Comment by HowdyHolto** [comment]
Im definitely planning to do several mods, cases and various key/encoder/etc designs once I have one. I will have to find someone in the group willing to build one for me (paid of course) so I can take measurements, understand the workflow, etc. I'm hoping to do some runs of custom encoder stalks, etc as well. I'm going to download all the CAD files this week and get started. You may have to point me to the measurements for any specific keys. I'm very aware of the custom keyboard space, but so far haven't dabbled in anything custom for it. I do have CAD experience however and several prototyping technologies in my shop. I would imagine given how popular the space has become for keyboards there are likely tweak able CAD files for all types of caps.

**Reply by stonepreston** [reply]
Regarding key cap models,  The keyV2 library is an openscad library for generating  mechanical key cap models and is what I used for the caps in the MCAD assembly. They don’t have a ton of profiles, but you could add one to that if you learn a bit of openscad

**Reply by HowdyHolto** [reply]
Oh very cool. I'll look into it asap. I'm normally working in Shapr3d and Fusion360 currently, but I'm hoping to take a look at openscad. I'm assuming it will open industry standard parametric/CAD filetypes. At least I hope so. I can really contribute if I'm able to work in my normal workflows for sure. Or perhaps if I open up any model someone experience in Openscad can convert them. I'll figure it out. Stoked!

**Reply by stonepreston** [reply]
Openscad is sort of like a CAD programming language that generates stl meshes. The learning curve is fairly steep but it’s not terrible once you get the hang of things.  The unfortunate thing is that it’s only for meshes, not solid geometry like step files. So to bring the models  into FreeCAD you have to convert the stl mesh to a solid which isn’t terrible, but just something to keep in mind

**Reply by HowdyHolto** [reply]
Oh I do .stl ALL the time. Ok, that's helpful. Then it shouldn't be a problem at all. I can also make STEP/OBJ/etc available to those who are able to use them.

**Comment by HowdyHolto** [comment]
FYI, I can also make us versions of the keycaps with debossed icons in them as well as custom shapes for different functions for navigating by "feel". Will be interested to get ideas once folks have their builds done and we see how things flow for everyone. I can do some usability research at that point if needed. 

I can print prototypes of the caps on FDM or resin, as well as mill them on the CNC. I have a fiber laser as well if we wanted to engrave existing keys. 

I'll likely start with using your current icons and just having the Openscad file available for everyone.

**Reply by stonepreston** [reply]
I’ve experimented with 3-D printing the key caps. I had the best results printing the key caps with icons embossed, and printing them icon side down on the print bed. This means curved profiles don’t work as well as the top of the key comes out flat. 

It definitely works, and with the keyV2 library it’s relatively easy to do. But it also just doesn’t look as good as cheap blanks with labels imo 

resin might be the best way to go

**Reply by stonepreston** [reply]
It would add some cool factor. Feel free to add to the key caps repository if you want to add printable keycaps. The current models there are only used in the FreeCAD assembly.

**Reply by HowdyHolto** [reply]
Yeah, good insight. I was thinking resin was likely the most like retail. Later this year I will be hopefully receiving an injection mold machine so I could experiment with variations that are more like mass market. Just gotta get there. Nice thing with resin in the short term is that we can also do transparent that works like a light guide if we add RGB capability. I have 2 Saturn printers and two Mars, so in theory, once things are dialed in I could produce small prototype sets in low runs for testing.

I also have a vacuum former that we could use for some funky decksaver/case ideas without having to print or mill every single thing. I need to check measurements and make sure I can fit the entire size on the platen. 

I love the idea of modeling a lemon cap that can be feel only and is always yellow. hahaha. or maybe a little lemon embedded in it.

**Reply by stonepreston** [reply]
@HowdyHolto dang you have a ton of cool tools

**Reply by HowdyHolto** [reply]
@stonepreston thanks. There's lots I'm hoping I can do for the community once my day job levels off. We've been going through a merger and it's intense ATM. I have a lot of goodies we can use to try things out. I'm in the process of packing up the workshop in town temporarily because I'm building a shop in the back of the house. We should be able to try a lot of things after that. If there is anything we want to physically fabricate for prototyping or small runs, I can likely do it. It's all a digital fabrication slant, so lot's of CNC stuff that can take files from the community with little overhead hopefully. I should have a small bench top injection molder as well this fall, which... Key caps and encoder knobs anyone? hahaha.
